lib/modules/platform/codecommit/codecommit-client.ts

```diff
@@ -306,37 +293,47 @@ export async function createPrApprovalRule(
 }
 
 export function getCodeCommitUrl(
-  region: string,
-  repoName: string,
-  credentials: Credentials
+  repoMetadata: RepositoryMetadata,
+  repoName: string
 ): string {
-  const signer = new aws4.RequestSigner(
-    {
-      service: 'codecommit',
-      host: `git-codecommit.${region}.amazonaws.com`,
-      method: 'GIT',
-      path: `v1/repos/${repoName}`,
-    },
-    credentials
-  );
+  logger.debug('get code commit url');
+  if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+    if (repoMetadata.cloneUrlHttp) {
+      return repoMetadata.cloneUrlHttp;
+    }
+    // shouldn't reach here, but just in case
+    return `https://git-codecommit.${
+      process.env.AWS_REGION ?? 'us-east-1'
+    }.amazonaws.com/v1/repos/${repoName}`;
+  }
```

Comment:
This section is executed in case IAM Roles are in use?

Reply:
Yes.

For normal user authentication we use aws signature V4 for http authentication (clone)

That is not possible without credentials so we can give the normal URL no credentials
And set the environment with git-credential-helper see docs.